src/createDispatcher.js

```diff
@@ -14,7 +14,7 @@ export default function createDispatcher(store, middlewares = []) {
     }
 
     const finalMiddlewares = typeof middlewares === 'function' ?
-      middlewares(getState) :
+      middlewares({ getState, dispatch }) :
```

Comment:
Isn't the idea to pass the full dispatch middleware chain instead of the default fn?